loadtest/user/userentity/websocket.go

```diff
@@ -239,9 +254,47 @@ func (ue *UserEntity) SendTypingEvent(channelId, parentId string) error {
 	if !ue.connected {
 		return errors.New("user is not connected")
 	}
-	ue.wsTyping <- userTypingMsg{
-		channelId,
-		parentId,
+	ue.dataChan <- userTypingMsg{
+		channelId: channelId,
+		parentId:  parentId,
+	}
+	return nil
+}
+
+func (ue *UserEntity) UpdateActiveChannel(channelId string) error {
+	if !ue.connected {
+		return errors.New("user is not connected")
+	}
+	ue.dataChan <- channelPresenceMsg{
+		channelId: channelId,
+	}
+	return nil
+}
+
+func (ue *UserEntity) UpdateActiveThread(channelId string) error {
+	if !ue.connected {
+		return errors.New("user is not connected")
+	}
+	// We don't really have a notion of RHS thread vs global thread in the load test.
+	// We either load a channel or load a thread. For now, we just set `is_thread_view`
+	// as both true and false to set the scope.
+	ue.dataChan <- threadPresenceMsg{
+		channelId:  channelId,
+		threadView: true,
+	}
+	ue.dataChan <- threadPresenceMsg{
+		channelId:  channelId,
+		threadView: false,
+	}
```

Comment:
The webapp has 2 different ways of loading a thread. One from RHS and one from global thread viewer. That means one can have 2 threads open at the same time. Since we don't simulate that, we are approximating it here.

Reply:
See https://github.com/mattermost/mattermost/pull/25794 for more details. And also the comment at the end of this section: https://mattermost.atlassian.net/wiki/spaces/~5de4a29d752c1b0d114f23c9/pages/2556788794/Rearchitecting+WebSocket+Events#Add-concept-of-%E2%80%9Cactive_channel%E2%80%9C-and-%E2%80%9Cactive_team%E2%80%9C-to-a-connection

Reply:
Ah, I get it now, we're simply opening both threads, right? That makes sense, thanks!

Reply:
Basically. Actually we are making the API call to open just a single thread. But we are setting the scope such that both threads are open.